returnn/tf/layers/basic.py

```diff
@@ -607,6 +607,59 @@ def get_out_data_from_opts(cls, sources, name, **kwargs):
     return get_concat_sources_data_template(sources, name="%s_output" % name)
 
 
+class NormLayer(_ConcatInputLayer):
+  """
+  Normalize over specified axes, e.g. time and/or feature axis.
+  In case of just feature, this corresponds to layer norm.
+  """
+  layer_class = "norm"
+
+  def __init__(self, axes, param_shape="F", scale=True, bias=True, epsilon=1e-6, **kwargs):
+    """
+    :param str|list[str] axes: axes over which the mean and variance are computed, e.g. "F" or "TF"
+    :param str|list[str]|int|list[int] param_shape: shape of the scale and bias parameters
+    :param bool scale: add trainable scale parameters
+    :param bool bias: add trainable bias parameters
+    :param float epsilon: epsilon for numerical stability
+    """
+    super(NormLayer, self).__init__(**kwargs)
+    assert not self.input_data.sparse
+    x = self.input_data.placeholder
+    dim = []
```

Comment:
`dim` is a confusing name. Call it `param_shape`. I.e. keep using `param_shape` and just adapt it.